src/context/AppProvider/AppContext.tsx

```diff
@@ -192,64 +190,121 @@ export const AppProvider = ({ children }: { children: React.ReactNode }) => {
+          // This allows every run of AccountIds per chain/accountNumber to run in parallel vs. all sequentally, so
+          // we can run each chain's side effects immediately
+          const accountNumberAccountIdsPromises = Object.values(
+            accountNumberAccountIdChunks(accountIds),
+          ).map(async accountIds => {
+            const results = await Promise.allSettled(
+              accountIds.map(async id => {
+                const result = await dispatch(getAccount.initiate({ accountId: id }))
+                return result
+              }),
+            )
+
+            results.forEach((res, idx) => {
+              if (res.status === 'rejected') return
+
+              const { data: account } = res.value
+              if (!account) return
+
+              const accountId = accountIds[idx]
+              const { chainId } = fromAccountId(accountId)
+
+              const { hasActivity } = account.accounts.byId[accountId]
+
+              const accountNumberHasChainActivity = !isUtxoChainId(chainId)
+                ? hasActivity
+                : // For UTXO AccountIds, we need to check if *any* of the scriptTypes have activity, not only the current one
+                  // else, we might end up with partial account data, with only the first 1 or 2 out of 3 scriptTypes
+                  // being upserted for BTC and LTC
+                  results.some((res, _idx) => {
+                    if (res.status === 'rejected') return false
+                    const { data: account } = res.value
+                    if (!account) return false
+                    const accountId = accountIds[_idx]
+                    const { chainId: _chainId } = fromAccountId(accountId)
+                    if (chainId !== _chainId) return false
+                    return account.accounts.byId[accountId].hasActivity
+                  })
+
+              // don't add accounts with no activity past account 0
+              if (accountNumber > 0 && !accountNumberHasChainActivity) {
+                chainIdsWithActivity = chainIdsWithActivity.filter(_chainId => _chainId !== chainId)
+                delete accountMetadataByAccountId[accountId]
+              } else {
+                // unique set to handle utxo chains with multiple account types per account
+                chainIdsWithActivity = Array.from(new Set([...chainIdsWithActivity, chainId]))
+
+                dispatch(portfolio.actions.upsertPortfolio(account))
+                const chainIdAccountMetadata = Object.entries(accountMetadataByAccountId).reduce(
+                  (acc, [accountId, metadata]) => {
+                    const { chainId: _chainId } = fromAccountId(accountId)
+                    if (chainId === _chainId) {
+                      acc[accountId] = metadata
+                    }
+                    return acc
+                  },
+                  {} as AccountMetadataById,
+                )
```

Comment:
[preferably blocking]: Untested but if possible this would be nicer:
```suggestion
                const chainIdAccountMetadata = {
                  [chainId]: accountMetadataByAccountId[toAccountId({ chainId, account })]
                }
```

Reply:
81b6e9b40fc813fcd56b424bd355dfdad66249f1 and confirmed we're still happy! https://jam.dev/c/6625fa90-5f30-4cd1-9acb-b46a9ad82dad

Reply:
Left as-is as this will lose the intent of this (keeping only the AccountId*s* - note plural here - for the current ChainId) and won't generalize to UTXOs (this would only keep the last AccountId for UTXO ChainIds)